vcd/resource_vcd_catalog_access_control.go

```diff
@@ -237,3 +319,59 @@ func resourceVcdCatalogAccessControlImport(_ context.Context, d *schema.Resource
 
 	return []*schema.ResourceData{d}, nil
 }
+
+// runWithRetry allows to run a given operation and retry it in case of error until the timeout is reached
+// * operationDescription is a human understandable description of the operation
+// * errorMessage is what to say in case of final error
+// * timeout is for how long we retry in case of failure
+// * preRun is an (optional) operation to run before attempting the operation
+// * operation is the main operation we are running
+func runWithRetry(operationDescription, errorMessage string, timeout time.Duration, preRun func() error, operation func() (any, error)) (any, error) {
```

Comment:
This could be a public function in some utility file, or close to the query operations?

Reply:
Since it is only used within this repo, I think it should stay private so far.
If we need to use it in other contexts, we can move it to govcd and make it public.